src/sentry/profiles/flamegraph.py

```diff
@@ -87,17 +90,59 @@ def get_span_intervals(
             "organization_id": organization_id,
         },
     )
-    return raw_snql_query(
+    data = raw_snql_query(
         request,
         referrer=Referrer.API_STARFISH_PROFILE_FLAMEGRAPH.value,
     )["data"]
+    spans_interval = []
+    for row in data:
+        start_ns = (int(datetime.fromisoformat(row["start_timestamp"]).timestamp()) * 10**9) + (
+            row["start_ms"] * 10**6
+        )
+        end_ns = (int(datetime.fromisoformat(row["end_timestamp"]).timestamp()) * 10**9) + (
+            row["end_ms"] * 10**6
+        )
+        interval = {}
+        interval["transaction_id"] = row["transaction_id"]
+        interval["start_ns"] = str(start_ns)
+        interval["end_ns"] = str(end_ns)
+        spans_interval.append(interval)
+    return spans_interval
+
+
+def get_span_intervals_from_nodestore(
+    project_id: str,
+    span_group: str,
+    transaction_ids: List[str],
+) -> List[Dict[str, Any]]:
+    spans_interval = []
+    for id in transaction_ids:
+        nodestore_event = eventstore.backend.get_event_by_id(project_id, id)
+        data = nodestore_event.data
+        for span in data.get("spans", []):
+            if span["hash"] == span_group:
```

Comment:
I'm not sure I understand what do you mean here.

This is meant to be an alternative backend to retrieve the same data that we're fetching with the indexed spans (in the context of starfish workflow) so the logic shouldn't diverge from the other backend in how we fetch the data.

Currently they're only using `span group`, not span op. This, as we discussed, has some limitations that need to and will be addressed (if description is empty, etc. etc.), but I don't think we should make this change here otherwise we end up with a different logic/workflow depending on the backend.


If we feel this should be addressed soon rather than later, I'd rather coordinate this with the starfish team so that we don't diverge the 2 implementations.

Reply:
I understand though this means if we were to this UI to the span summary pages, we wouldn't be able to rely on the results as it may contain frames from other spans with the same group but different op